tutorials/colors/colors.py

```diff
@@ -5,59 +5,85 @@
++--------------------------------------+--------------------------------------+
+| Case-insensitive color name from     | - ``'xkcd:sky blue'``                |
+| `xkcd color survey`_ with ``'xkcd:'``| - ``'xkcd:eggshell'``                |
+| prefix                               |                                      |
++--------------------------------------+--------------------------------------+
+| Case-insensitive Tableau Colors from | - 'tab:blue'                         |
+| 'T10' categorical palette            | - 'tab:orange'                       |
+|                                      | - 'tab:green'                        |
+|                                      | - 'tab:red'                          |
+|                                      | - 'tab:purple'                       |
+| .. note:: This is the default color  | - 'tab:brown'                        |
+|           cycle.                     | - 'tab:pink'                         |
+|                                      | - 'tab:gray'                         |
+|                                      | - 'tab:olive'                        |
+|                                      | - 'tab:cyan'                         |
++--------------------------------------+--------------------------------------+
+| "CN" color spec where ``'C'``        | - ``'C0'``                           |
+| precedes a number acting as an index | - ``'C1'``                           |
+| into the default property cycle      +--------------------------------------+
+|                                      | :rc:`axes.prop_cycle`                |
+| .. note:: Indexing occurs at         |                                      |
+|           rendering time and defaults|                                      |
+|           to black if cycle does not |                                      |
+|           inlcude color.             |                                      |
++--------------------------------------+--------------------------------------+
 
 .. _xkcd color survey: https://xkcd.com/color/rgb/
 
-"Red", "Green", and "Blue" are the intensities of those colors, the combination
-of which span the colorspace.
+"Red", "Green", and "Blue" are the intensities of those colors. In combination,
+they represent the colorspace.
 
-How "Alpha" behaves depends on the ``zorder`` of the Artist.  Higher
-``zorder`` Artists are drawn on top of lower Artists, and "Alpha" determines
-whether the lower artist is covered by the higher.
-If the old RGB of a pixel is ``RGBold`` and the RGB of the
-pixel of the Artist being added is ``RGBnew`` with Alpha ``alpha``,
-then the RGB of the pixel is updated to:
-``RGB = RGBOld * (1 - Alpha) + RGBnew * Alpha``.  Alpha
-of 1 means the old color is completely covered by the new Artist, Alpha of 0
-means that pixel of the Artist is transparent.
+"Alpha" depends on the ``zorder`` of the Artist.  Matplotlib draws higher
```

Comment:
reading through this whole section, I'm wondering if it'd be easier to explain via diagrams/mpl patches (and understand if this should be pitched to another issue/pr?) like two overlapping boxes - A1, A2 ... with A1 > A2 when z_a1>z_a2 and vice versa.  Then make a second example where alpha on A1 is very low, and possibly third example where A1 and A2 both have like .5 alpha.

Reply:
It's a good explanation and I think it's a valuable example to illustrate. Though I'm not entirely sure if it would make the most sense here in the colors tutorials section. Is there already a place with `zorder` information where this could be referenced? Or, is this something that could be a new section?

Reply:
the zorder demo doesn't include color, https://matplotlib.org/gallery/misc/zorder_demo.html#sphx-glr-gallery-misc-zorder-demo-py
diagrams about color/zorder interaction: https://colab.research.google.com/drive/1U6I8yICGWvwwgOIgr7C_Hg7uoCDDD4OH#scrollTo=HesrKyf98bSX

Reply:
I would not say "alpha depends on zorder" as it is the two of them together that determine how the colors are blended.  Maybe something like:

> Alpha controls how the colors are combined when Artists overlap each other.  The Artists are rendered in order based on their zorder, Artists with higher zorder are "on top of" Artist with lower zorder, and combined via alpha compositing.    To compute the result of blending a new artist we use the equation ::
>   
>     RGB_{out} = RGB_{bottom} * (1 - \alpha) + RGB_{top} * \alpha
>
> If $\alpha$ is 1 the top color fully occludes the bottom color, if $\alpha$ is 0 the top color is not visible and they are blended at for intermediate values of $\alpha$.  Because each artist is blended into the cumulative result of all previous artists, re-ordering Artists is not commutative. 

xref: https://en.wikipedia.org/wiki/Alpha_compositing  ?

Reply:
^ I found that a *bit* confusing because "bottom" really means whatever is bellow the current artist, which itself may be the result of alpha compositing.

```
Artists are composed based on their zorder, followed by the order they are added to an axes.  The Alpha for an artist controls how the RGB color of the new artist is combined with the colors already on the axes using the equation ::
   RGB_{new} = RGB_{below} * (1-\alpha) + RGB_{artist} * \alpha
If $\alpha$ is 1 the top color fully occludes the bottom color, if $\alpha$ is 0 the top color is not visible, and they are blended for intermediate values.  
```
I find the last suggested sentence more mysterious than helpful in that I'm not sure what situation they could be confused for being commutative?

Reply:
I like @jklymak 's suggested names s better than mine.

I have a strong memory of cranking through the algebra to convince my self that it was non-commutative at one point so I am not sure it is completely obvious ;).   We have at least one bug report about this (https://github.com/matplotlib/matplotlib/issues/4862).

Reply:
OK, I agree.  I wonder if this is the place to go into it in a bit more detail or not, but if so, it should be a paragraph with an example.

Reply:
I've changed the language of Alpha and Artists based on `zorder` for the section, though I'm not entirely sure how to include an example of the colors changing. Would placing the code sample from @story645 below the text be the best fit?

Reply:
I dunno that you need the full code sample (which maybe stash in a file for reference & link to in docs). I'd just use the images to illustrate the idea (+/- a change to the title).

Reply:
Should I add a static file to the library and reference it in the paragraph? Also, does "Specifying Colors" need to change for the title or is the title change something else?

Reply:
So yeah I think a static image would be fine...and my title change I meant the axes titles in the figure if they're not as clear as could be.